### PR TITLE
Upgrade wire up

### LINKED ISSUE #3339: Finish commit-upgrade work

### PR DESCRIPTION
**Pull request by Fraser999**
Several changes and fixes to support upgrading nodes.

Closes #3339.

### CODE REVIEW

File: node/src/components/block_synchronizer.rs [hunk 1]
```diff
@@ -317,8 +318,8 @@ impl BlockSynchronizer {
             let action = builder.block_acquisition_action(rng);
             let peers = action.peers_to_ask(); // pass this to any fetcher
             let need_next = action.need_next();
+            debug!("BlockSynchronizer: needs next: {:?}", need_next);
```
Comment (EdHastingsCasperAssociation) [inline node/src/components/block_synchronizer.rs]: trace! ?

File: node/src/components/block_synchronizer/block_builder.rs [hunk 2]
```diff
@@ -325,7 +329,7 @@ impl BlockBuilder {
                     self.promote_peer(maybe_peer);
                 }
                 Ok(FinalitySignatureAcceptance::Noop) => {
-                    // noop
+                    self.touch();
```
Comment (EdHastingsCasperAssociation) [inline node/src/components/block_synchronizer/block_builder.rs:332]: we may not need to touch here after all, given the other fix we made later in need_next itself to check the inflight latch

File: node/src/reactor/main_reactor/control.rs [hunk 3]
```diff
@@ -137,10 +175,66 @@ impl MainReactor {
...
+                        };
+                        (DELAY_BEFORE_SHUTDOWN, effects)
+                    }
+                    None => (
+                        Duration::ZERO,
+                        utils::new_shutdown_effect(
+                            "validator_weights cannot be missing in Upgrade state",
+                        ),
+                    ),
+                }
```
Comment (EdHastingsCasperAssociation) [inline node/src/reactor/main_reactor/control.rs:237]: i'll extract this as part of a general clean up after this PR and my upcoming sync-to-historical PR both merge

**Review by EdHastingsCasperAssociation — APPROVED** [review]

File: node/src/components/block_synchronizer/block_builder.rs (hunk 2 above)
Comment (Fraser999) [reply node/src/components/block_synchronizer/block_builder.rs:332]: I'll check again once your fixes are merged.

**Pull request merged**